[CI] MultiClusterSpecIT test {date.Convert no zone} failing

**Build Scans:**
- [elasticsearch-periodic #7878 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/e7s54i3mcyzbw)
- [elasticsearch-periodic #7860 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/2rce4za2xnhd6)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-clusters:v8.13.4#oldToNew" -Dtests.class="org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT" -Dtests.method="test {date.Convert no zone}" -Dtests.seed=11C2B69FAD063454 -Dtests.bwc=true -Dtests.locale=he-Hebr-IL -Dtests.timezone=Asia/Gaza -Druntime.java=24
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bdate.Convert%20no%20zone%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected an empty list
0: <unexpected> but was "Line 2:13: evaluation of [TO_DATETIME(str)] failed, treating result as null. Only first 20 failures recorded."
1: <unexpected> but was "Line 2:13: java.lang.IllegalArgumentException: failed to parse date field [2025-04-11T05:00:00.000] with format [yyyy-MM-dd'T'HH:mm:ss.SSS'Z']"
```

**Issue Reasons:**
- [8.19] 2 failures in test test {date.Convert no zone} (0.9% fail rate in 231 executions)
- [8.19] 2 failures in step 8.13.4_bwc (50.0% fail rate in 4 executions)
- [8.19] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/date.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/date.csv-spec
@@ -200,7 +200,7 @@ birth_date:date         |bd:date
 1964-06-02T00:00:00.000Z|1964-06-02T00:00:00.000Z
 ;
 
-convert with zones
+convert with zones#[skip:-8.13.99, reason: default date formatter is changed in 8.14]
 ROW str = "2025-04-11T05:00:00+0400"
 | EVAL dt = TO_DATETIME(str);
 
@@ -216,7 +216,7 @@ dt:datetime              | bool:boolean
 2025-04-11T01:00:00.000Z | true
 ;
 
-convert no zone
+convert no zone#[skip:-8.13.99, reason: default date formatter is changed in 8.14]
 ROW str = "2025-04-11T05:00:00.000"
 | EVAL dt = TO_DATETIME(str);
 

